Manager/ElasticSearchManager.php

```diff
@@ -12,75 +12,77 @@
 namespace Ferdyrurka\CommandBus\Manager;
 
 use Elasticsearch\Client;
-use Elasticsearch\ClientBuilder;
-use Ferdyrurka\CommandBus\DependencyInjection\Database\ElasticSearchDatabase;
-use Ferdyrurka\CommandBus\DependencyInjection\Parameters;
-use Symfony\Component\DependencyInjection\ContainerInterface;
+use Ferdyrurka\CommandBus\Exception\EmptyEntityException;
+use Ferdyrurka\CommandBus\Util\ElasticSearch\ElasticSearchConnection;
+use Ferdyrurka\CommandBus\Util\ElasticSearch\ReflectionEntity;
 
 /**
  * Class ElasticSearchManager
  * @package Ferdyrurka\CommandBus\Util
  */
-class ElasticSearchManager implements ElasticSearchManagerInterface
+class ElasticSearchManager implements ManagerInterface
 {
     /**
-     *
+     * @var ElasticSearchConnection
      */
-    private const PREFIX = Parameters::PREFIX . '_' . ElasticSearchDatabase::DATABASE_NAME . '_';
+    protected $elasticSearchConnection;
 
     /**
-     * @var ContainerInterface
+     * @var Client
      */
-    protected $container;
+    protected $client;
 
     /**
-     * ElasticSearchManager constructor.
-     * @param ContainerInterface $container
+     * @var string
      */
-    public function __construct(ContainerInterface $container)
-    {
-        $this->container = $container;
-    }
+    protected $index;
 
     /**
-     * @return Client
+     * @var array
      */
-    public function getManager(): Client
-    {
+    protected $persists;
 
-        return $this->createClient();
+    /**
+     * ElasticSearchManager constructor.
+     * @param ElasticSearchConnection $elasticSearchConnection
+     */
+    public function __construct(ElasticSearchConnection $elasticSearchConnection)
+    {
+        $this->elasticSearchConnection = $elasticSearchConnection;
     }
 
     /**
-     * @return string
+     * @param object $entity
      */
-    public function getIndex(): string
+    public function persist(object $entity): void
     {
-        return $this->container->getParameter(self::PREFIX . 'index');
+        $this->persists[] = $entity;
     }
 
     /**
-     * @return Client
+     * @throws EmptyEntityException
+     * @throws \ReflectionException
      */
-    protected function createClient(): Client
+    public function flush(): void
```

Comment:
Flush dla porównania w doctrine przyjmuje encje i wtedy tylko ja zapisuje do bazy a nie cały unit of work

Reply:
Rozumiem, dzięki! :)